core/modules/search/templates/_results_normal.inc.php

```diff
@@ -148,7 +148,7 @@
                         <?php echo (!$issue->hasEstimatedTime()) ? '-' : \thebuggenie\core\entities\Issue::getFormattedTime($issue->getEstimatedTime(true, true)); ?>
                     </td>
                     <td class="sc_spent_time<?php if (!$issue->hasSpentTime()): ?> faded_out<?php endif; ?>"<?php if (!in_array('spent_time', $visible_columns)): ?> style="display: none;"<?php endif; ?>>
-                        <?php echo (!$issue->hasSpentTime()) ? '-' : \thebuggenie\core\entities\Issue::getFormattedTime($issue->getSpentTime(true, true)); ?>
+                        <?php echo (!$issue->hasSpentTime() || !$issue->isSpentTimeVisibleByPermission()) ? '-' : \thebuggenie\core\entities\Issue::getFormattedTime($issue->getSpentTime(true, true)); ?>
```

Comment:
Into my solution, SpentTime can be hidden by permission although may contain more values, so otherwise from other fields, i can't use same logic.
I think that a better solution may be keep the method hasTimeSpent() with actual logic, and add the new logic (permission check) into isSpentTimeVisible() as
<
public function isSpentTimeVisible()
        {
            return (bool) ($this->isFieldVisible('spent_time') || ($this->hasSpentTime() && $this->getProject()->canSeeTimeSpent()));
        }
>
in place of mine isSpentTimeVisibleByPermission()